.circleci/config.yml

```diff
@@ -1,15 +1,90 @@
 version: 2.1
 
+commands:
+  e2e-command:
+    description: "e2e test"
+    steps:
+      - run:
+          name: Update PATH and Define Environment Variable at Runtime
+          command: |
+            echo 'export GOPATH=/home/circleci/go' >> $BASH_ENV
+            echo 'export PATH=$GOPATH/bin:/usr/local/go/bin:$PATH' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: setup golang
+          command: |
+            sudo rm -rf /usr/local/go
+            GO_VERSION=1.13.4
+            curl -fsL https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | sudo tar xzf - -C /usr/local
+      - run:
+          name: Setup
+          command: |
+            make setup
+            cd e2e
+            make setup
+      - run:
+          name: Run test
+          command: |
+            cd e2e
+            make launch-kind test
+
 jobs:
   build:
     docker:
-    - image: quay.io/cybozu/golang:1.13-bionic
+      - image: quay.io/cybozu/golang:1.13-bionic
     working_directory: /work
     steps:
-    - checkout
+      - checkout
+      - run: make setup SUDO=""
+      - run:
+          name: Confirm manifests and api code were generated
+          command: |
+            make manifests generate
+            diffs=$(git diff --name-only)
+            if [ "$diffs" != "" ]; then printf "%s\n" "$diffs"; exit 1; fi
+      - run: make test
+      - run: make moco-controller
+  image:
+    docker:
+      - image: docker:stable
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: cp -a /tmp/workspace build
+      - setup_remote_docker
+      - run: docker build . -t quay.io/cybozu/myso:latest
+      - run:
+          name: Push docker image to Quay.io
+          command: |
+            if [ "$CIRCLE_TAG" = "" ]; then exit 0; fi
+            docker login -u $QUAY_USER -p $QUAY_PASSWORD quay.io
+            echo "pushing image ..."
+            TAG=$(echo $CIRCLE_TAG | cut -c 2-)
+            docker tag quay.io/cybozu/myso:latest quay.io/cybozu/myso:$TAG
+            docker push quay.io/cybozu/myso:$TAG
+            if echo $TAG | grep -q -e - ; then
+                echo ===== Skip pushing branch tags for pre-release $TAG =====
+                exit 0
+            fi
+            BRANCH=$(echo $TAG | cut -d "." -f 1-2)
+            docker tag quay.io/cybozu/myso:latest quay.io/cybozu/myso:$BRANCH
+            docker push quay.io/cybozu/myso:$BRANCH
+  e2e:
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: large
+    working_directory: /home/circleci/work
+    steps:
+      - checkout
+      - e2e-command
 
 workflows:
   version: 2
   main:
     jobs:
       - build
+      - image:
+          requires:
+            - build
```

Comment:
Dockerfile seems not to use the built binary, doesn't it?

Reply:
As you wrote, Dockerfile does not use any output of `build`.  `build` does not produce anything as persistent objects.
But I still want to put it in `requires` because `build` runs tests and it works as a requisite.  I also want to put `e2e` in the `requires` list.